matrix-connection.c

```diff
@@ -287,23 +303,16 @@ static void _join_failed(MatrixConnectionData *conn,
 void matrix_connection_join_room(struct _PurpleConnection *pc,
         const gchar *room, GHashTable *components)
 {
-    GHashTable *copy;
-    GHashTableIter iter;
-    gpointer key, value;
+    GHashTable *newcomponents;
 
     MatrixConnectionData *conn = purple_connection_get_protocol_data(pc);
 
-    /* we have to copy the components table, so that we can pass it back
-     * later on :/
-     */
-    copy = g_hash_table_new_full(g_str_hash, g_str_equal, g_free, g_free);
-    g_hash_table_iter_init (&iter, components);
-    while (g_hash_table_iter_next (&iter, &key, &value)) {
-        g_hash_table_insert(copy, g_strdup(key), g_strdup(value));
-    }
+    /* Assume old components is corrupt */
+    newcomponents = g_hash_table_new_full(g_str_hash, g_str_equal, NULL, g_free);
+    g_hash_table_insert(newcomponents, PRPL_CHAT_INFO_ROOM_ID, g_strdup(room));
```

Comment:
I'm still unclear why you want to change this. We used to take a copy of the `components` table, so that `_join_failed` could pass it to `purple_serv_got_join_chat_failed` without worrying about exactly what was supposed to be in there.

Please explain why this is better?